osu.Game.Rulesets.Catch/UI/Catcher.cs

```diff
@@ -479,33 +454,109 @@ private void updateState(CatcherAnimationState state)
+            caughtObject.X = source.X - X;
+            caughtObject.IsOnPlate = true;
+
+            caughtObject.Anchor = Anchor.TopCentre;
+            caughtObject.Origin = Anchor.Centre;
+            caughtObject.Scale *= 0.5f;
+            caughtObject.LifetimeStart = source.StartTime;
+            caughtObject.LifetimeEnd = double.MaxValue;
+
+            PlaceOnPlate(caughtObject);
+
+            if (!caughtObject.StaysOnPlate)
+                Explode(caughtObject);
+        }
+
+        private DrawablePalpableCatchHitObject createCaughtObject(PalpableCatchHitObject source)
+        {
+            switch (source)
             {
-                fruit.Anchor = Anchor.TopLeft;
-                fruit.Position = caughtFruitContainer.ToSpaceOfOtherDrawable(fruit.DrawPosition, ExplodingFruitTarget);
+                case Banana banana:
+                    return new DrawableBanana(banana);
+
+                case Fruit fruit:
+                    return new DrawableFruit(fruit);
 
-                if (!caughtFruitContainer.Remove(fruit))
-                    // we may have already been removed by a previous operation (due to the weird OnLoadComplete scheduling).
-                    // this avoids a crash on potentially attempting to Add a fruit to ExplodingFruitTarget twice.
-                    return;
+                case TinyDroplet tiny:
+                    return new DrawableTinyDroplet(tiny);
 
-                ExplodingFruitTarget.Add(fruit);
+                case Droplet droplet:
+                    return new DrawableDroplet(droplet);
+
+                default:
+                    return null;
             }
+        }
+
+        private void clearPlate(DroppedObjectAnimation animation)
+        {
+            var caughtObjects = caughtFruitContainer.Children.ToArray();
+            caughtFruitContainer.Clear(false);
+
+            droppedObjectTarget.AddRange(caughtObjects);
+
+            foreach (var caughtObject in caughtObjects)
+                drop(caughtObject, animation);
+        }
+
+        private void removeFromPlate(DrawablePalpableCatchHitObject caughtObject, DroppedObjectAnimation animation)
+        {
+            if (!caughtFruitContainer.Remove(caughtObject))
+                throw new InvalidOperationException("Can only drop a caught object on the plate");
+
+            droppedObjectTarget.Add(caughtObject);
 
-            var actionTime = Clock.CurrentTime;
+            drop(caughtObject, animation);
+        }
+
+        private void drop(Drawable d, DroppedObjectAnimation animation)
+        {
+            var originalX = d.X * Scale.X;
 
-            fruit.ApplyCustomUpdateState += onFruitOnApplyCustomUpdateState;
-            onFruitOnApplyCustomUpdateState(fruit, fruit.State.Value);
+            d.Anchor = Anchor.TopLeft;
+            d.Position = caughtFruitContainer.ToSpaceOfOtherDrawable(d.DrawPosition, droppedObjectTarget);
+
+            animate(d, animation, originalX);
```

Comment:
I believe you're going to need to re-add these animations after a skin change (since an `updateState` is forced by `DrawableHitObject` on skin change):

![2020-12-03 17 57 15](https://user-images.githubusercontent.com/191335/100986818-0aa44580-3591-11eb-9b0c-3de1629e3c0a.gif)


You can test this easily using the "random skin" hotkey (Ctrl+Shift+R)